### PR TITLE
Pass addon object to plugin load method

### PR DESCRIPTION
**Pull request by QbDesu**
This way the plugin can read it's own name and version from it's metadata which can help with deduplication. and libraries don't need to rely on getting them via an object during plugin initialization, or rather no object provided and coded in the plugin itself, and without having the library parse the metadata from the plugin file by itself.

Also includes getting rid of some hardcoded strings.

### CODE REVIEW

File: renderer/src/modules/pluginmanager.js [hunk 1]
```diff
@@ -80,12 +80,12 @@ export default new class PluginManager extends AddonManager {
             const PluginClass = addon.exports;
             const thePlugin = new PluginClass();
             addon.instance = thePlugin;
-            addon.name = thePlugin.getName ? thePlugin.getName() : addon.name || "No name";
-            addon.author = thePlugin.getAuthor ? thePlugin.getAuthor() : addon.author || "No author";
-            addon.description = thePlugin.getDescription ? thePlugin.getDescription() : addon.description || "No description";
-            addon.version = thePlugin.getVersion ? thePlugin.getVersion() : addon.version || "No version";
+            addon.name = thePlugin.getName ? thePlugin.getName() : addon.name || ;
```
Comment (Inve1951) [inline renderer/src/modules/pluginmanager.js:83]: that's an unexpected token

Comment (QbDesu) [reply renderer/src/modules/pluginmanager.js:83]: oops yeah, forgot to stage that after fixing it

Comment (QbDesu) [reply renderer/src/modules/pluginmanager.js]: pushed the fix, tested it locally with a plugin like this
```js
/**
 * @name foo
 * @version bar
 * @some_meta_flag_without_value
 * @some_meta_field test
 */
module.exports = class {
    load(addon) {
        console.log(addon);
    }

    start(){

    }
}
```

**Comment by rauenzi** [comment]
I'm not certain of the utility of this but off the top of my head, should definitely not be passing the object reference but a copy of the information needed.

**Comment by Inve1951** [comment]
Not sure I agree. Going thru existing code i see that `BdApi.Plugins.get(id)` and relatives return the real thing as-well.
If we only pass an excerpt/copy then the plugin would later have to call `BdApi.Plugins.get(copy.id)` to retrieve a reference which could be avoided.

GC can handle the circular reference should the plugin choose to keep one on the instance so I don't see an issue.

What's your reasoning for not wanting to expose the raw `addon` here? @rauenzi

**Comment by Inve1951** [comment]
Regarding utility of this: It allows plugins to use the data provided in the file's meta and obliterates the need to keep a copy in code or use string literals for things like `BdApi.saveData(addonName, key, data)` because then it can re-use `addon.name`

**Comment by rauenzi** [comment]
I think that API should also not return a reference. I don't see a real need for plugins to access that.

**Comment by rauenzi** [comment]
Reasoning would be because that structure is meant for BD internals and allowing plugins to modify it freely would mean more sanity checking internally and the point of something like this structure is to avoid that internally

**Comment by Inve1951** [comment]
How about we wrap them with a get-only recursive Proxy? Are there scenarios where a plugin has legitimate reason to directly modify such an object?

**Comment by QbDesu** [comment]
I like the idea of using a recursive readonly proxy. While at it we could wrap the result of other endpoints like Plugins.get in that proxy as well, as I don't think there is any legitamate use of modifying the addon object from the plugin side.
While looking at the code I also noticed another thing that we may want to change. Currently the addon object directly contains all the parsed metadata fields. This means a plugin could some stupid stuff like `@instance foo`   or things like that which may be overridden at a later stage but I think it would be wise to prevent eventualities like that. I would suggest putting the metadata in it's old field called `metadata` and copying the needed fields over to the root addon object.

**Comment by rauenzi** [comment]
I still don't see the utility of this being passed to `load()`

**Comment by rauenzi** [comment]
This has been decided against in favor of passing to the wrapping function/constructor

**Pull request closed without merging**